zcash_primitives/src/transaction/sighash_v5.rs

```diff
@@ -73,41 +112,52 @@ fn transparent_input_sigdigests<A: transparent::Authorization>(
     //   c. value of the output spent by this input (8-byte little endian)
     //   d. nSequence of the input (4-byte little endian)
     let mut ch = hasher(ZCASH_TRANSPARENT_INPUT_HASH_PERSONALIZATION);
-    let txin = &bundle.vin[input.index()];
-    txin.prevout.write(&mut ch).unwrap();
-    input.script_code().write(&mut ch).unwrap();
-    ch.write_all(&input.value().to_i64_le_bytes()).unwrap();
-    ch.write_u32::<LittleEndian>(txin.sequence).unwrap();
+    if let SignableInput::Transparent {
+        index,
+        script_pubkey,
+        value,
+        ..
+    } = input
+    {
+        let txin = &bundle.vin[*index];
+        txin.prevout.write(&mut ch).unwrap();
+        ch.write_all(&value.to_i64_le_bytes()).unwrap();
+        script_pubkey.write(&mut ch).unwrap();
+        ch.write_u32::<LittleEndian>(txin.sequence).unwrap();
```

Comment:
The field order here is correct, though, it's just the comment above that is wrong.